compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1805,6 +1805,13 @@ rustc_queries! {
         eval_always
         desc { "alloc error handler kind for the current crate" }
     }
+    /// Which allocator kind to codegen. If `Some(_)` this will be the same as
+    /// `allocator_kind`, but it may return `None` in more cases (e.g. if using
+    /// allocator definitions from a dylib dependency).
+    query codegen_allocator_kind(_: ()) -> Option<AllocatorKind> {
+        eval_always
```

Comment:
This `eval_always` can be removed as the query doesn't depend on untracked state but only on other queries (`dependency_formats` and `allocator_kind`). I'm not sure whether this makes sense to have as a query in the first place or whether it should just be a normal helper function. I think a normal function would make more sense as the output of this will usually change when the inputs (`dependency_formats` and `allocator_kind`) change and it's also pretty cheap to compute and only called twice.

Reply:
Maybe compiler/rustc_codegen_ssa/src/base.rs? It doesn't quite fit, but I don't know a better place.

Reply:
OK, I moved the logic to a function there